block/internal/syncing/da_retriever.go

```diff
@@ -55,6 +61,9 @@ func NewDARetriever(
 		logger:          logger.With().Str("component", "da_retriever").Logger(),
 		namespaceBz:     coreda.NamespaceFromString(config.DA.GetNamespace()).Bytes(),
 		namespaceDataBz: coreda.NamespaceFromString(config.DA.GetDataNamespace()).Bytes(),
+		pendingHeaders:  make(map[uint64]*types.SignedHeader),
+		pendingData:     make(map[uint64]*types.Data),
+		headerDAHeights: make(map[uint64]uint64), // Track DA height for each header
```

Comment:
what if multiple headers are included in the same da height? this is a valid case as we submit in chunks

Reply:
checked, and the key is the header height for this is fine.
i will add a test however.